File: CSharp/ModelObjects.cs
using System;
using System.Collections.Generic;

namespace export
{
    public class StoreEvent : Product
    {
        public StoreEvent(string name, string type, Store store, Price price) : base(name, type, 0, price)
        {
            SetLocation(store);
        }

        private Store Store { get; set; }

        public void SetLocation(Store store)
        {
            Store = store;
            store.AddStockedItems(this);
        }

        public override bool IsEvent()
        {
            return true;
        }
    }

    public class Product
    {
        public Product(string name, string type, int weight, Price price)
        {
            Name = name;
            Id = type;
            Weight = weight;
            Price = price;
        }

        public string Name { get; set; }
        public string Id { get; set; }
        public int Weight { get; set; }
        public Price Price { get; set; }

        public virtual bool IsEvent()
        {
            return false;
        }
    }

    public class Price
    {
        public Price(double amount, string currency)
        {
            Amount = amount;
            Currency = currency;
        }

        public double Amount { get; set; }
        public string Currency { get; set; }

        public double GetAmountInCurrency(string currency)
        {
            return Amount;
        }
    }

    public class Store
    {
        public readonly IList<Product> Stock = new List<Product>();
        public string Name { get; set; }
        public string Id { get; set; }


        public void AddStockedItems(params Product[] products)
        {
            foreach (var product in products) Stock.Add(product);
        }
    }

    public class Order
    {
        public readonly IList<Product> Products = new List<Product>();

        public string Id { get; set; }
        public DateTime Date { get; set; }
        public Store Store { get; set; }

        public double TotalDollars()
        {
            return 0;
        }
    }
}

File: Python/model_objects.py

class ModelObject(object):
    def save_to_database(self):
        raise Exception("You shouldn't call save_to_database from a unit test")


class Order(ModelObject):
    def __init__(self, id, date, store, products):
        self.id = id
        self.date = date
        self.store = store
        self.products = products[:]

    def total_dollars(self):
        dollars = 0
        for product in self.products:
            dollars += product.price.get_amount_in_currency("USD")
        return dollars


class Price(ModelObject):
    def __init__(self, amount, currency):
        self.amount = amount
        self.currency = currency

    def get_amount_in_currency(self, currency):
        return self.amount


class Product(ModelObject):
    def __init__(self, name, id, weight, price):
        self.name = name
        self.id = id
        self.weight = weight
        self.price = price
        self.isEvent = False


class Store(ModelObject):
    """Represents a physical Store where you can go and buy products and attend events."""
    def __init__(self, name, id, products=None):
        self.name = name
        self.id = id
        self.stock = products or []


class StoreEvent(Product):
    """Ticket to In-store event, eg makeover, eyeshadow masterclass
    or beauty product launch evening reception"""
    def __init__(self, name, store, id, price):
        super(StoreEvent, self).__init__(name, id, 0, price)
        self.store = store
        self.isEvent = True
